python/lsst/obs/hsc/makeTransmissionCurves.py

```diff
@@ -0,0 +1,153 @@
+    if atMax is None:
+        atMax = throughput[-1]
+    return TransmissionCurve.makeSpatiallyConstant(throughput=throughput, wavelengths=wavelengths,
+                                                   throughputAtMin=atMin, throughputAtMax=atMax)
+
+
+def getOpticsTransmission():
+    """Return a dictionary of TransmissionCurves describing the combined
+    throughput of HSC and the Subaru primary mirror.
+
+    Dictionary keys are string dates (YYYY-MM-DD) indicating the beginning of
+    the validity period for the curve stored as the associated dictionary
+    value.  If the curve is spatially varying, it will be defined in focal
+    plane coordinates.
+
+    Dictionary values may be None to indicate that no TransmissionCurve is
+    valid after the date provided in the key.
+    """
+    mirror2010 = readTransmissionCurveFromFile("M1-2010s.txt", unit="nm")
+    camera = readTransmissionCurveFromFile("throughput_popt2.txt")
+    camera *= readTransmissionCurveFromFile("throughput_win.txt")
+    return {
+        HSC_BEGIN: mirror2010*camera,
+        "2017-10-01": None   # mirror recoating begins, approximately
+    }
+
+
+def getSensorTransmission():
+    """Return a nested dictionary of TransmissionCurves describing the
+    throughput of each sensor.
+
+    Outer directionary keys are string dates (YYYY-MM-DD), with values
+    a dictionary mapping CCD ID to TransmissionCurve.  If the curve
+    is spatially varying, it will be defined in pixel coordinates.
+
+    Dictionary values may be None to indicate that no TransmissionCurve is
+    valid after the date provided in the key.
+    """
+    qe = readTransmissionCurveFromFile("qe_ccd_HSC.txt", atMin=0.0, atMax=0.0)
+    return {HSC_BEGIN: {n: qe for n in range(116)}}
+
+
+def getAtmosphereTransmission():
+    """Return a dictionary of TransmissionCurves describing the atmospheric
+    throughput at Mauna Kea.
+
+    Dictionary keys are string dates (YYYY-MM-DD) indicating the beginning of
+    the validity period for the curve stored as the associated dictionary
+    value.  The curve is guaranteed not to be spatially-varying.
+
+    Dictionary values may be None to indicate that no TransmissionCurve is
+    valid after the date provided in the key.
+    """
+    average = readTransmissionCurveFromFile("modtran_maunakea_am12_pwv15_binned10ang.dat")
+    return {HSC_BEGIN: average}
+
+
+def getFilterTransmission():
+    """Return a nested dictionary of TransmissionCurves describing the
+    throughput of each HSC filter.
+
+    Outer directionary keys are string dates (YYYY-MM-DD), with values
+    a dictionary mapping filter name to TransmissionCurve.  If the curve
+    is spatially varying, it will be defined in pixel coordinates.
+
+    Filter curve names are in the long form used as data ID values (e.g.
+    'HSC-I').
+
+    Dictionary values may be None to indicate that no TransmissionCurve is
+    valid after the date provided in the key.
+    """
+    g = {}
```

Comment:
Can you use a better name than `g`? `filterTraces`, maybe?

Reply:
I'll call it `module`, since it's a dict of the module scope of the `.py` file.